packages-next/keystone/src/session/index.ts

```diff
@@ -74,10 +74,10 @@ type FieldSelections = {
   - [ ] We could support additional where input to validate item sessions (e.g an isEnabled boolean)
 */
 
-export function withItemData<T extends { listKey: string; itemId: string }>(
-  createSession: () => SessionStrategy<T>,
+export function withItemData<U extends Record<string, any>>(
+  createSession: () => SessionStrategy<U>,
   fieldSelections: FieldSelections = {}
-): () => SessionStrategy<T & { data: any }> {
+): () => SessionStrategy<U & { listKey: string; itemId: string; data: any }> {
```

Comment:
This function being generic seems unnecessary. You could replace U and and the `U & { ... }` with passing in some type which is `{ listKey: string; itemId: string; data: any }`

Reply:
I'm not sure what you mean by this 🤔

Reply:
I'm saying remove the `<U extends Record<string, any>>` generic